snow/engine/common/error.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package common
+
+import "fmt"
+
+var (
+	_ error = (*AppError)(nil)
```

Comment:
nit: we moved to value receivers, so this should be `_ error = AppError{}`

Reply:
Since `AppError` is intended to be used as an `error` (which is an interface) I think it makes sense for us to use a pointer here

Reply:
We're back to pointer receiver, hence resolving this